axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetGenerationServiceImpl.java

```diff
@@ -114,6 +115,16 @@ public FixedAsset generateAndComputeLines(FixedAsset fixedAsset) throws AxelorEx
       generateAndComputeFiscalFixedAssetLines(fixedAsset);
       generateAndComputeFixedAssetDerogatoryLines(fixedAsset);
       generateAndComputeIfrsFixedAssetLines(fixedAsset);
+
+      if (fixedAsset.getFixedAssetCategory() != null
+          && fixedAsset.getFixedAssetCategory().getPeriodicityTypeSelect() != null
```

Comment:
You don't need to check if an integer can be null

Reply:
Resolved